### PR TITLE
ENH: Output thickness, curvature, and sulcal depth files

### PR DESCRIPTION
**Pull request by mgxd**
Continues #296

### CODE REVIEW

**Comment by mgxd** [comment]
@effigies do you happen to know the BIDS compliant suffixes for `sulc`/`thickness`/`curv` files? I can expand our patterns (https://github.com/nipreps/niworkflows/blob/master/niworkflows/data/nipreps.json) but want to avoid making incompatible outputs from the jump

**Comment by effigies** [comment]
https://bids-specification.readthedocs.io/en/bep011/05-derivatives/04-structural-derivatives.html#surface-mapped-anatomical-scalar-derivatives

**Comment by mgxd** [comment]
Perfect, staying true to its FS equivalent 😄 

Thanks, ready for review

File: smriprep/workflows/surfaces.py [hunk 1]
```diff
@@ -532,9 +537,17 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+    surfmorph_list = pe.Node(
+        niu.Merge(3, ravel_inputs=True),
+        name="surfmorph_list",
+        run_without_submitting=True,
+    )
+    fscurv2funcgii = pe.MapNode(
+        MRIsConvertData(out_datatype="gii", to_scanner=True, target_surface="smoothwm"),
+        iterfield="scalarcurv_file", name="fscurv2funcgii",
     )
-    fix_surfs = pe.MapNode(NormalizeSurf(), iterfield="in_file", name="fix_surfs")
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:537]: Why are we dropping `fix_surfs`? See #295,

File: smriprep/workflows/surfaces.py [hunk 2]
```diff
@@ -552,9 +565,12 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
         (surface_list, fs2gii, [('out', 'in_file')]),
-        (fs2gii, fix_surfs, [('converted', 'in_file')]),
-        (inputnode, fix_surfs, [('fsnative2t1w_xfm', 'transform_file')]),
-        (fix_surfs, outputnode, [('out_file', 'surfaces')]),
+        (get_surfaces, surfmorph_list, [('thickness', 'in1'),
+                                        ('sulc', 'in2'),
+                                        ('curv', 'in3')]),
+        (surfmorph_list, fscurv2funcgii, [('out', 'scalarcurv_file')]),
+        (fs2gii, outputnode, [('converted', 'surfaces')]),
+        (fscurv2funcgii, outputnode, [('converted', 'morphometrics')]),
```
Comment (effigies) [inline smriprep/workflows/surfaces.py:573]: Bringing back what I think were mistaken deletions...

```suggestion
        (fs2gii, fix_surfs, [('converted', 'in_file')]),
        (inputnode, fix_surfs, [('fsnative2t1w_xfm', 'transform_file')]),
        (fix_surfs, outputnode, [('out_file', 'surfaces')]),
        (get_surfaces, surfmorph_list, [('thickness', 'in1'),
                                        ('sulc', 'in2'),
                                        ('curv', 'in3')]),
        (surfmorph_list, fscurv2funcgii, [('out', 'scalarcurv_file')]),
        (fscurv2funcgii, outputnode, [('converted', 'morphometrics')]),
```

**Review by effigies — COMMENTED** [review]
Looks right. Have you run and made sure that the data arrays in the output GIFTIs correspond to the correct FreeSurfer curvature files?

**Comment by mgxd** [comment]
I have yet to run this on my own data

File: smriprep/workflows/surfaces.py (hunk 1 above)
Comment (effigies) [reply smriprep/workflows/surfaces.py:537]: No. These are updates to the coordinates themselves. Curvature files are just scalar maps.

**Comment by mgxd** [comment]
CIFTIs generated by MRIsConvertData match their FS equivalents (loaded with `nibabel.freesurfer.io.read_morph_data`)

File: smriprep/workflows/surfaces.py [hunk 3]
```diff
@@ -532,9 +538,18 @@ def init_gifti_surface_wf(*, name="gifti_surface_wf"):
...
+        fs.MRIsConvert(out_datatype="gii", to_scanner=True), iterfield="in_file", name="fs2gii",
     )
     fix_surfs = pe.MapNode(NormalizeSurf(), iterfield="in_file", name="fix_surfs")
+    surfmorph_list = pe.Node(
+        niu.Merge(3, ravel_inputs=True),
+        name="surfmorph_list",
+        run_without_submitting=True,
+    )
+    fscurv2funcgii = pe.MapNode(
+        MRIsConvertData(out_datatype="gii", to_scanner=True, target_surface="smoothwm"),
```
Comment (mgxd) [inline smriprep/workflows/surfaces.py:550]: Lingering question: Should use the `?h.white` or `?h.smoothwm` surface files?

Comment (effigies) [reply smriprep/workflows/surfaces.py:550]: I don't think `to_scanner` or `target_surface` will make any difference here.

**Review by effigies — APPROVED** [review]
@mgxd If you want to experiment with modifying/removing some args from `fscurv2funcgii` (and actually, that name isn't great, either) that's fine with me. Also okay with merging as-is.

**Pull request merged**